src/catalog/src/system_schema/utils.rs

```diff
@@ -12,48 +12,32 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-pub mod tables;
-
-use std::sync::{Arc, Weak};
+use std::sync::Weak;
 
-use common_config::Mode;
 use common_meta::key::TableMetadataManagerRef;
-use common_procedure::ProcedureManagerRef;
-use meta_client::client::MetaClient;
 use snafu::OptionExt;
 
 use crate::error::{Result, UpgradeWeakCatalogManagerRefSnafu};
+use crate::information_schema::InformationExtensionRef;
 use crate::kvbackend::KvBackendCatalogManager;
 use crate::CatalogManager;
 
-/// Try to get the server running mode from `[CatalogManager]` weak reference.
-pub fn running_mode(catalog_manager: &Weak<dyn CatalogManager>) -> Result<Option<Mode>> {
-    let catalog_manager = catalog_manager
-        .upgrade()
-        .context(UpgradeWeakCatalogManagerRefSnafu)?;
-
-    Ok(catalog_manager
-        .as_any()
-        .downcast_ref::<KvBackendCatalogManager>()
-        .map(|manager| manager.running_mode())
-        .copied())
-}
+pub mod tables;
 
-/// Try to get the `[MetaClient]` from `[CatalogManager]` weak reference.
-pub fn meta_client(catalog_manager: &Weak<dyn CatalogManager>) -> Result<Option<Arc<MetaClient>>> {
+/// Try to get the `[InformationExtension]` from `[CatalogManager]` weak reference.
+pub fn information_extension(
+    catalog_manager: &Weak<dyn CatalogManager>,
+) -> Result<Option<InformationExtensionRef>> {
```

Comment:
Seems the caller of `information_extension` always needs the not-none one (by `.context(GetInformationExtensionSnafu)?` the result). If that's the case, I suggest return `Result<InformationExtensionRef>` instead.